### DISCUSSION TITLE
SLEAP root error ConcatOp unequal shapes

### DISCUSSION THREAD

**Discussion by frr665** [bonsai-rx/bonsai#1312] [Q&A] [answered]
Hi bonsai users, 

I've been trying to use SLEAP with bonsai for a closed loop experiment. Unfortunately I'm totally stuck with the following error (bellow). 

I've used the sleap-export command line to export my model into protocol buffer format. I've checked that I used the config file corresponding to the exported model. 

Does someone has an idea of what happens and how to fix this ? 

Thanks a lot for your help, 

Hugo 

![image](https://user-images.githubusercontent.com/117995542/228799410-5d7e914d-e96c-44a6-adcc-9359bbe07088.png)

I have attached bellow a link with my test video, my model .pd file and my json config file if you want to replicate this issue. 

https://drive.google.com/file/d/1zL0c2FJq49C7mDv6b5EKDOau5Q7en5gK/view?usp=sharing

**Comment by bruno-f-cruz** [comment]
It seems the shape of you input image doesn't match the input of the network. I can you check in the json files that resulted from the export what is the expected input?

**Reply by frr665** [reply]
Thanks for your answer !
In the info.json file, the expected input size seems to be the following : (None, 1066, 1386, 1)
Should I use a resize node after my file capture node to fix that ? 


Content of the info.json file : 
```
{
    "frozen_model_inputs": [
        "Tensor(\"x:0\", shape=(None, 1066, 1386, 1), dtype=uint8)"
    ],
    "frozen_model_outputs": [
        "Tensor(\"Identity:0\", shape=(None, 1, None), dtype=float32)",
        "Tensor(\"Identity_1:0\", shape=(None, 1, None, 2), dtype=float32)"
    ],
    "model_structured_input_signature": [
        [],
        {
            "input_1": "TensorSpec(shape=(None, 1066, 1386, 1), dtype=tf.uint8, name='input_1')"
        }
    ],
    "model_structured_outputs_signature": {
        "instance_peak_vals": "TensorSpec(shape=(None, 1, None), dtype=tf.float32, name='instance_peak_vals')",
        "instance_peaks": "TensorSpec(shape=(None, 1, None, 2), dtype=tf.float32, name='instance_peaks')"
    }
}
```

**Reply by bruno-f-cruz** [reply]
What is the size of your input images?

**Reply by frr665** [reply]
1228x870

**Comment by bruno-f-cruz** [comment, marked as answer]
Ideally, you would use the same input as the one you used for training. But
ye, you can use resize or resizecanvas to change the input image resolution.

**Reply by frr665** [reply]
Ok I resized it to 1386x1066 and it worked ! Thanks a lot !!

![image](https://user-images.githubusercontent.com/117995542/228837889-2a5dceb0-4618-43a1-912d-cfbdcc8f1b16.png)

**Reply by bruno-f-cruz** [reply]
I would advise you to set the offset property to 0,0 so the coordinates of the tracked parts do not change relative to the original image.
